MLIR.cmake

```diff
@@ -139,7 +142,14 @@ function(add_onnx_mlir_library name)
     )
 
   if (ARG_LINK_LIBS)
-    target_link_libraries(${name} ${ARG_LINK_LIBS})
+    if (ARG_LINK_TARGET_ARCH_LIBS)
+      llvm_map_components_to_libnames(TARGET_ARCH_LIBS AllTargetsDescs AllTargetsInfos AllTargetsCodeGens AllTargetsAsmParsers MC)
```

Comment:
Does this not set `TARGET_ARCH_LIBS` in such a way that it can just be passed to `target_link_libraries` below?

Reply:
yes you are right, fixed. Good catch !